src/Connection/connection.js

```diff
@@ -38,12 +46,44 @@ class Connection extends Component {
 
   state = {
     loading: false,
+    parityInstallLocation: true,
     token: '',
     validToken: false
   }
 
+  componentDidMount () {
+    if (isElectron()) {
+      const { ipcRenderer, remote } = electron;
+      const parityInstallLocation = remote.getGlobal('parityInstallLocation');
+
+      this.setState({ parityInstallLocation });
+
+      // Run parity if parityInstallLocation !== null and not connected yet
+      if (!parityInstallLocation) { return; }
+
+      // After 3s, check if ui is still isConnecting
+      // If yes, then try to run `parity`
+      // The reason why we do this after 3s, is that even when parity is
+      // running, isConnecting is true on componentWillMount (lag to ping the
+      // node). -Amaury 13.03.2018
+      // TODO Find a more reliable way to know if parity is running or not
+      setTimeout(() => {
+        if (!this.props.isConnecting) { return; }
+        console.log('Launching parity.');
+        ipcRenderer.send('asynchronous-message', 'run-parity');
+      }, 3000);
```

Comment:
I'm doing something a bit hacky here. On UI launch, isConnecting is true (while pinging the node). I'm saying here that, if after 3s, the ui is still isConnecting, then I assume that the parity node is down, so I launch parity.

Possible bug: if because of some lag here and there, the ping takes more than 3s, then there might be 2 instances of parity running at some point. Though the second one would close because of ws port conflict.

@jacogr maybe you have a better idea on how to do this?

Reply:
No :( Unless we start listing running processes.